src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -828,6 +823,52 @@ fn generate_targets(
         }
     }
 
+    if mode.is_doc_scrape() {
```

Comment:
Not a blocker. Do you feel it better if we extract this piece of code to a separate function doing post-process after calling `generate_targets`? Doing so avoids adding more parameters on `generate_targets`. 

Don't blindly agree with me, though 😆.

Reply:
The issue is that this code needs to be in the middle of `generate_targets` --- it post-processes the *proposals* but not the *units*. 

In my proposed refactor, I'm going to separate out `generate_targets` into `TargetGenerator::make_proposals` and `TargetGenerator::proposals_to_units`. Then I can cleanly insert the doc-scrape-specific code after `make_proposals` as post-processing.